wagtail/images/models.py

```diff
@@ -1384,16 +1384,35 @@ def get_upload_to(self, filename):
     def check(cls, **kwargs):
         errors = super().check(**kwargs)
         if not cls._meta.abstract:
-            if not any(
+            in_unique_together = any(
                 set(constraint) == {"image", "filter_spec", "focal_point_key"}
                 for constraint in cls._meta.unique_together
-            ):
+            )
+
+            # We only include fields to check if they are a UniqueConstraint, and
+            # they have no conditions on them.
+            unique_constraint_fields = [
+                constraint.fields
+                for constraint in cls._meta.constraints
+                if (
+                    isinstance(constraint, UniqueConstraint)
+                    and constraint.condition is None
+                )
+            ]
+            in_unique_constraint = any(
+                set(constraint) == {"image", "filter_spec", "focal_point_key"}
+                for constraint in unique_constraint_fields
+            )
```

Comment:
I think this can be simplified as a single-pass loop like the following:

```suggestion
        in_unique_constraint = any(
            isinstance(constraint, models.UniqueConstraint)
            and constraint.condition is None
            and set(constraint.fields) == {"image", "filter_spec", "focal_point_key"}
            for constraint in cls._meta.constraints
        )
```